### PR TITLE
APL-1679 added ResourceLocator;

### PR DESCRIPTION
**Pull request by alzinchenko**
fixed importing the genesis block from config dir by chainId;

### CODE REVIEW

**Review by oleksiylukin — APPROVED** [review]

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java [hunk 1]
```diff
@@ -404,24 +425,44 @@ private int savePublicKeys(final TransactionalDataSource dataSource) {
...
         log.debug("Genesis accounts json resource path = " + path);
+        final InputStream is = resourceLocator.locate(path).orElseThrow(() -> new RuntimeException("The resource could not be found, path=" + path));
+
+        final Queue<Map.Entry<String, Long>> sortedEntries = loadGenesisAccountsFromIS(is);
 
+        final int balanceNumber = sortedEntries.size();
+        validateBalanceNumber(balanceNumber);
+
+        return sortedEntries.stream()
+            .skip(1) //skip first account to collect only genesis accounts
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java:448]: What does happen on 'skip(1)' if 'sortedEntries' is empty ? may be something like .orElseThrow(...) is needed for such case?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java [hunk 2]
```diff
@@ -432,25 +473,21 @@ private int savePublicKeys(final TransactionalDataSource dataSource) {
                 }
             }
         } catch (IOException e) {
-            throw new RuntimeException("Failed to load genesis accounts", e);
+            log.error("Failed to load genesis accounts, cause:{}", e.getMessage());
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java:476]: Previous variant generates stack trace into error log, but new one doesn't.

**Review by yuriy-larin-firstbridge — APPROVED** [review]

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java (hunk 2 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java:476]: next line throws GenesisImportException with stack trace

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java (hunk 1 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java:448]: There is always at least one entry.

**Review by AndrewBoyarsky — APPROVED** [review]

**Pull request merged**